secant/Dependencies/DerivationTool/DerivationToolTestKey.swift

```diff
@@ -6,58 +6,26 @@
-        deriveUnifiedAddressFromUnifiedViewingKey:
-            XCTUnimplemented("\(Self.self).deriveUnifiedAddressFromUnifiedViewingKey", placeholder: TestUnifiedAddress()),
-        deriveTransparentAddressFromPublicKey: XCTUnimplemented("\(Self.self).deriveTransparentAddressFromPublicKey", placeholder: ""),
-        deriveTransparentPrivateKey: XCTUnimplemented("\(Self.self).deriveTransparentPrivateKey", placeholder: ""),
-        deriveTransparentAddressFromPrivateKey: XCTUnimplemented("\(Self.self).deriveTransparentAddressFromPrivateKey", placeholder: ""),
-        isValidExtendedViewingKey: XCTUnimplemented("\(Self.self).isValidExtendedViewingKey", placeholder: false),
+        deriveUnifiedViewingKeyFromSpendingKey: XCTUnimplemented("\(Self.self).deriveUnifiedViewingKeyFromSpendingKey"),
+        deriveSpendingKey: XCTUnimplemented("\(Self.self).deriveSpendingKey"),
         isValidTransparentAddress: XCTUnimplemented("\(Self.self).isValidTransparentAddress", placeholder: false),
-        isValidShieldedAddress: XCTUnimplemented("\(Self.self).isValidShieldedAddress", placeholder: false),
+        isValidSaplingAddress: XCTUnimplemented("\(Self.self).isValidShieldedAddress", placeholder: false),
         isValidZcashAddress: XCTUnimplemented("\(Self.self).isValidZcashAddress", placeholder: false)
     )
 }
 
-extension DerivationToolClient {
-    struct TestUnifiedAddress: UnifiedAddress {
-        var tAddress: ZcashLightClientKit.TransparentAddress
-        var zAddress: ZcashLightClientKit.SaplingShieldedAddress
-        
-        init(tAddress: ZcashLightClientKit.TransparentAddress = "", zAddress: ZcashLightClientKit.SaplingShieldedAddress = "") {
-            self.tAddress = tAddress
-            self.zAddress = zAddress
-        }
-    }
-}
-
 extension DerivationToolClient {
     static let noOp = Self(
-        deriveViewingKeys: { _, _ in [] },
-        deriveViewingKey: { _ in "" },
-        deriveSpendingKeys: { _, _ in [] },
-        deriveShieldedAddress: { _, _ in "" },
-        deriveShieldedAddressFromViewingKey: { _ in "" },
-        deriveTransparentAddress: { _, _, _ in "" },
-        deriveUnifiedViewingKeysFromSeed: { _, _ in [] },
-        deriveUnifiedAddressFromUnifiedViewingKey: { _ in TestUnifiedAddress() },
-        deriveTransparentAddressFromPublicKey: { _ in "" },
-        deriveTransparentPrivateKey: { _, _, _ in "" },
-        deriveTransparentAddressFromPrivateKey: { _ in "" },
-        isValidExtendedViewingKey: { _ in false },
-        isValidTransparentAddress: { _ in false },
-        isValidShieldedAddress: { _ in false },
-        isValidZcashAddress: { _ in false }
+        deriveUnifiedViewingKeyFromSpendingKey: { _ in throw NSError(domain: "NotImplemented", code: 0, userInfo: nil) },
+        deriveSpendingKey: { _, _ in throw NSError(domain: "NotImplemented", code: 0, userInfo: nil) },
```

Comment:
Do we want to use NSError here? We already have String: Error conformance, so this could be simplified to throw "NotImplemented".

Reply:
Changed